### PR TITLE
Dr 794 change password

### PR DESCRIPTION
**Pull request by mvillaseco**
Changes: 
 - Add Change password functionality
 - Add Ctrl
 - Add service
 - Add view (to be refactored)
 - Add style (to be refactored)

### CODE REVIEW

File: src/wwwroot/controllers/MyProfileCtrl.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    ]);
+    vm.changePassword = changePassword;
+    vm.updateValidation = updateValidation;
+
+    function updateValidation(newPass, confNewPass) {
+      if (!$scope.form.pass.$modelValue || !$scope.form.confPass.$modelValue) {
+        $scope.form.confPass.$setValidity('same', null);
+      } else if ($scope.form.pass.$modelValue != $scope.form.confPass.$modelValue) {
+        $scope.form.confPass.$setValidity('same', false);
+      } else {
```
Comment (pbarrios) [inline src/wwwroot/controllers/MyProfileCtrl.js]: If in all cases you're setting a value you can avoid this extra else:

```javascript
 function updateValidation(newPass, confNewPass) {
   $scope.form.confPass.$setValidity('same', true);
   if (!$scope.form.pass.$modelValue || !$scope.form.confPass.$modelValue) {
     $scope.form.confPass.$setValidity('same', null);
    } else if ($scope.form.pass.$modelValue != $scope.form.confPass.$modelValue) {
     $scope.form.confPass.$setValidity('same', false);
    }
  }
```

File: src/wwwroot/locales/es-translation.js [hunk 2]
```diff
@@ -262,5 +262,11 @@
   "domain_manager_dkim_spf_status_th": "SPF & DKIM",
   "domain_manager_status_error_tooltip": "El DKIM no está configurado aún.",
   "domain_manager_status_warning_tooltip": "El SPF no está configurado aún.",
-  "select_choose_status": "Selecciona estado"
+  "select_choose_status": "Selecciona estado",
+  "submenu_my_profile": "Mi Perfil",
+  "my_profile_title": "Mi Perfil",
+  "my_profile_subtitle": "Password",
+  "my_profile_old_password":"Old Password",
```
Comment (pbarrios) [inline src/wwwroot/locales/es-translation.js]: Don't you have to translate these sentences to spanish?

File: src/wwwroot/services/auth.js [hunk 3]
```diff
@@ -197,5 +198,25 @@
...
+        data: {
+          "username": getUserName(),
+          "old_password": old_pass,
+          "password": newPass
+        }
+      })
+      .then(function (response) {
+        return response;
+      })
+      .catch(function (reason) {
```
Comment (pbarrios) [inline src/wwwroot/services/auth.js]: If the promise fails we're not doing anything. Is that OK? If so, why adding the catch function?

File: src/wwwroot/controllers/MyProfileCtrl.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    function changePassword(form) {
+      vm.submitted = true;
+      if (form.pass.$modelValue != form.confPass.$modelValue || !form.$valid) {
+        return;
+      }
+      auth.changePassword(form.oldPass.$modelValue, form.pass.$modelValue, $translate.use())
+      .then(function() {
+        vm.showChangePassContainer = false;
+        vm.changePasswordSuccess = true;
```
Comment (pbarrios) [inline src/wwwroot/controllers/MyProfileCtrl.js]: Is the change password always successful? Don't we have to validate the response from the server? Or to catch an exception thrown during the promise execution?

**Comment by pbarrios** [comment]
Thanks @mvillaseco! Is it possible to add some tests to this amazing work you have done?

File: src/wwwroot/controllers/MyProfileCtrl.js (hunk 1 above)
Comment (andresmoschini) [inline src/wwwroot/controllers/MyProfileCtrl.js]: I like @mvillaseco approach, is more like a _switch_, at least for me it is easier to read.

File: src/wwwroot/services/auth.js [hunk 5]
```diff
@@ -197,5 +198,25 @@
...
+          'Content-Type': 'application/json'
+        },
+        data: {
+          "username": getUserName(),
+          "old_password": old_pass,
+          "password": newPass
+        }
+      })
+      .then(function (response) {
+        return response;
```
Comment (andresmoschini) [inline src/wwwroot/services/auth.js]: This _then_ is redundant. Could you remove it?

File: src/wwwroot/styles/views/_settings.scss [hunk 6]
```diff
@@ -199,6 +199,109 @@
...
+    border-top: $suit-border-default;
+  }
+  .item:last-of-type {
+    padding-bottom: 2.5em;
+    border-bottom: $suit-border-default;
+  }
+  .password.item{
+    overflow: hidden;
+    input{align-items: flex-end;}
+    transition: height 0.3s ease;ight: toem(245px);
```
Comment (andresmoschini) [inline src/wwwroot/styles/views/_settings.scss]: typo?

File: src/wwwroot/services/auth.js (hunk 3 above)
Comment (mvillaseco) [inline src/wwwroot/services/auth.js]: Yeah, i'll remove the catch, and if the promise fails the error handler will catch that error.

**Pull request merged**